test/services/samples/metadata/file_import_service_test.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module Samples
+  module Metadata
+    class FileImportServiceTest < ActiveSupport::TestCase
+      def setup
+        @john_doe = users(:john_doe)
+        @jane_doe = users(:jane_doe)
+        @project = projects(:project1)
+        @sample1 = samples(:sample1)
+        @sample2 = samples(:sample2)
+
+        @csv = File.new('test/fixtures/files/metadata/valid.csv', 'r')
+      end
+
+      # bin/rails test test/services/samples/metadata/file_import_service_test.rb
+
+      test 'import sample metadata with permission' do
+        assert_authorized_to(:update_sample?, @project,
+                             with: ProjectPolicy,
+                             context: { user: @john_doe }) do
+          params = { file: @csv, sample_id_column: 'sample_name' }
+          Samples::Metadata::FileImportService.new(@project, @john_doe, params).execute
+        end
+      end
+
+      test 'import sample metadata without permission' do
+        assert_raises(ActionPolicy::Unauthorized) do
+          params = { file: @csv, sample_id_column: 'sample_name' }
+          Samples::Metadata::FileImportService.new(@project, @jane_doe, params).execute
+        end
+      end
+
+      test 'import sample metadata with empty params' do
+        assert_empty Samples::Metadata::FileImportService.new(@project, @john_doe, {}).execute
+      end
+
+      test 'import sample metadata via csv file' do
+        params = { file: @csv, sample_id_column: 'sample_name' }
+        response = Samples::Metadata::FileImportService.new(@project, @john_doe,
+                                                            params).execute
+        assert_equal({ @sample1.name => { updated: %w[metadatafield1 metadatafield2 metadatafield3], not_updated: [] },
+                       @sample2.name => { updated: %w[metadatafield1 metadatafield2 metadatafield3],
```

Comment:
These will need to be updated throughout the testing once [PR 339](https://github.com/phac-nml/irida-next/pull/339) is merged with the new hash values as mentioned in the previous comment.

Reply:
Please see 4685902c953494151b4408fb2b406c8e400d0bbd.